contracts/hooks/SupplyCapAllowedCallerIssuanceHook.sol

```diff
@@ -0,0 +1,159 @@
+ */
+contract SupplyCapAllowedCallerIssuanceHook is Ownable, IManagerIssuanceHook {
+    using SafeMath for uint256;
+    using AddressArrayUtils for address[];
+
+    /* ============ Events ============ */
+
+    event SupplyCapUpdated(uint256 _newCap);
+    event CallerStatusUpdated(address indexed _caller, bool _status);
+    event AnyoneCallableUpdated(bool indexed _status);
+    
+    /* ============ State Variables ============ */
+
+    // Cap on totalSupply of Sets
+    uint256 public supplyCap;
+
+    // Boolean indicating if anyone can call function
+    bool public anyoneCallable;
+
+    // Mapping of contract addresses allowed to call function
+    mapping(address => bool) public callAllowList;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Constructor, overwrites owner and original supply cap.
+     *
+     * @param _initialOwner      Owner address, overwrites Ownable logic which sets to deployer as default
+     * @param _supplyCap         Supply cap for Set (in wei of Set)
+     */
+    constructor(
+        address _initialOwner,
+        uint256 _supplyCap
+    )
+        public
+    {
+        supplyCap = _supplyCap;
+
+        // Overwrite _owner param of Ownable contract
+        transferOwnership(_initialOwner);
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * Adheres to IManagerIssuanceHook interface, and checks to make sure the current issue call won't push total supply over cap.
+     */
+    function invokePreIssueHook(
+        ISetToken _setToken,
+        uint256 _issueQuantity,
+        address _sender,
+        address /*_to*/
+    )
+        external
+        override
+    {
+        _validateAllowedContractCaller(_sender);
+        
+        uint256 totalSupply = _setToken.totalSupply();
+        require(totalSupply.add(_issueQuantity) <= supplyCap, "Supply cap exceeded");
+    }
+
+    /**
+     * Adheres to IManagerIssuanceHook interface
+     */
+    function invokePreRedeemHook(
+        ISetToken _setToken,
+        uint256 _redeemQuantity,
+        address _sender,
+        address _to
+    )
+        external
+        override
+    {}
+
+    /**
+     * ONLY OWNER: Updates supply cap
+     */
+    function updateSupplyCap(uint256 _newCap) external onlyOwner {
+        supplyCap = _newCap;
+        SupplyCapUpdated(_newCap);
+    }
+
+    /**
+     * ONLY OWNER: Toggle ability for passed addresses to call only allowed caller functions
+     *
+     * @param _callers           Array of caller addresses to toggle status
+     * @param _statuses          Array of statuses for each caller
+     */
+    function updateCallerStatus(address[] calldata _callers, bool[] calldata _statuses) external onlyOwner {
+        require(_callers.length == _statuses.length, "Array length mismatch");
+        require(_callers.length > 0, "Array length must be > 0");
+        require(!_callers.hasDuplicate(), "Cannot duplicate callers");
```

Comment:
We added [a set of helpers][1] to the `AddressArrayUtils` lib in set-protocol-v2 for GeneralIndexModule that make exactly these checks. They reduce this require block to...
```sol
_callers.validatePairsWithArray(_statuses);
``` 

Would it be helpful to port those over? I can open a PR for that today if so. 

[1]: https://github.com/SetProtocol/set-protocol-v2/blob/7657f9a9f77f8e639bb6650b1da5496ba2c4080b/contracts/lib/AddressArrayUtils.sol#L156-L224

Reply:
Yeah this would be helpful actually

Reply:
Those are ported in #43